lib/auth/rotate.go

```diff
@@ -481,116 +496,174 @@ func startNewRotation(req rotationReq, ca services.CertAuthority) error {
-		// pairs, and generates new CAs that are trusted, but not used in the cluster.
-		signingKeys = [][]byte{signingKeys[0], sshPrivPEM}
-		checkingKeys = [][]byte{checkingKeys[0], sshPubPEM}
-		keyPairs = []services.TLSKeyPair{keyPairs[0], *tlsKeyPair}
+		// Perform a length check because not every CA has all types of keys.
+		// For example, a JWT services.CertAuthority will not have SSH or TLS keys.
+		// Similarly SSH and TLS services.CertAuthority do not have JWT keys.
+		if len(signingKeys) > 0 {
+			signingKeys = [][]byte{signingKeys[0], sshPrivateKey}
+			checkingKeys = [][]byte{checkingKeys[0], sshPublicKey}
+		}
+		if len(tlsKeyPairs) > 0 {
+			tlsKeyPairs = []services.TLSKeyPair{tlsKeyPairs[0], *tlsKeyPair}
+		}
+		if len(jwtKeyPairs) > 0 {
+			jwtKeyPairs = []services.JWTKeyPair{jwtKeyPairs[0], jwtKeyPair}
+		}
+
 		rotation.State = services.RotationStateInProgress
 		rotation.Phase = services.RotationPhaseInit
 	}
 
+	// Update services.CertAuthority.
 	if err := ca.SetSigningKeys(signingKeys); err != nil {
 		return trace.Wrap(err)
 	}
 	if err := ca.SetCheckingKeys(checkingKeys); err != nil {
 		return trace.Wrap(err)
 	}
-	ca.SetTLSKeyPairs(keyPairs)
+	ca.SetTLSKeyPairs(tlsKeyPairs)
+	ca.SetJWTKeyPairs(jwtKeyPairs)
 	ca.SetRotation(rotation)
-	// caSigningAlg is only set when signing algorithm was explicitly set in
-	// the config file. When config file doesn't set a value, preserve the
-	// signing algorithm of the existing CA.
+	// The certificate signing algorithm is only set when signing algorithm is
+	// explicitly set in the config file. If the config file doesn't set a value,
+	// preserve the signing algorithm of the existing CA.
 	if req.caSigningAlg != nil {
 		ca.SetSigningAlg(*req.caSigningAlg)
 	}
+
 	return nil
 }
 
-// updateClients swaps old and new cert authorities:
-//
-// * old CAs exist and are trusted, but are not used for signing
-// * the new CAs are now used for signing
-// * remote components will reload with new certificates used for client connections
+// updateClients swaps old CA (index 0) and new CA (index 1).
 //
+// * Old CAs continue to be trusted, but are no longer used for signing.
+// * New CAs are used for signing.
+// * Remote components will reload with new certificates used for client
+//   connections.
 func updateClients(ca services.CertAuthority, mode string) error {
 	rotation := ca.GetRotation()
 
 	signingKeys := ca.GetSigningKeys()
 	checkingKeys := ca.GetCheckingKeys()
-	keyPairs := ca.GetTLSKeyPairs()
+	tlsKeyPairs := ca.GetTLSKeyPairs()
+	jwtKeyPairs := ca.GetJWTKeyPairs()
 
-	signingKeys = [][]byte{signingKeys[1], signingKeys[0]}
-	checkingKeys = [][]byte{checkingKeys[1], checkingKeys[0]}
+	// Perform a length check because not every CA has all types of keys.
+	// For example, a JWT services.CertAuthority will not have SSH or TLS keys.
+	// Similarly SSH and TLS services.CertAuthority do not have JWT keys.
+	if len(signingKeys) > 0 {
```

Comment:
Should the checks for swapping keys be `if len(keys) > 1`?
There should be at least 2 keys to swap them without a panic.